pkg/wireguard/agent/agent.go

```diff
@@ -468,10 +468,11 @@ func (a *Agent) deletePeerByPubKey(pubKey wgtypes.Key) error {
 // updatePeerByConfig updates the WireGuard kernel peer config based on peerConfig p
 func (a *Agent) updatePeerByConfig(p *peerConfig) error {
 	peer := wgtypes.PeerConfig{
-		PublicKey:         p.pubKey,
-		Endpoint:          p.endpoint,
-		AllowedIPs:        p.allowedIPs,
-		ReplaceAllowedIPs: true,
+		PublicKey:                   p.pubKey,
+		Endpoint:                    p.endpoint,
+		AllowedIPs:                  p.allowedIPs,
+		ReplaceAllowedIPs:           true,
+		PersistentKeepaliveInterval: &option.Config.WireguarPersistentKeepalive,
```

Comment:
So this will set `&time.Duration(0)` as the default persistent keepalive. I think we should leave this nil if the keepalive is unset (i.e. zero)

Reply:
Updated.